x/ccv/provider/keeper/key_assignment.go

```diff
@@ -102,6 +104,7 @@ func (k Keeper) IterateAllValidatorConsumerPubKeys(
 	}
 }
 
+// TODO: maybe make variadic DeleteValidatorConsumerPubKey(ctx sdk.Context, providerAddr sdk.ConsAddress, chainIDs ...string)
```

Comment:
Where would you use it? From where do you get the list of consumer chain IDs?

Reply:
There's this `keeper.IterateConsumerChains`.

I agree it is not needed. I was just making notes for myself